src/NetMQ/OutgoingSocketExtensions.cs

```diff
@@ -98,8 +200,182 @@ public static IOutgoingSocket SendMore([NotNull] this IOutgoingSocket socket, [N
 
         #endregion
 
+        #endregion
+
+        #region Sending a multipart message as byte arrays
+
+        #region Blocking
+
+        public static void SendMultipartBytes([NotNull] this IOutgoingSocket socket, params byte[][] frames)
+        {
+            SendMultipartBytes(socket, (IEnumerable<byte[]>)frames);
+        }
+
+        public static void SendMultipartBytes([NotNull] this IOutgoingSocket socket, IEnumerable<byte[]> frames)
+        {
+            var enumerator = frames.GetEnumerator();
+
+            // move to the first emlement, if false frames is empty
+            if (!enumerator.MoveNext())
+            {
+                throw new ArgumentException("frames is empty", "frames");
+            }
+
+            var current = enumerator.Current;
+
+            // we always one item back to make sure we send the last frame without the more flag
+            while (enumerator.MoveNext())
+            {
+                // this is a more frame
+                socket.SendMoreFrame(current);
+
+                current = enumerator.Current;
+            }
+
+            // sending the last frame
+            socket.SendFrame(current);
+        }
+
+        #endregion
+
+        #region Timeout
+
+        public static bool TrySendMultipartBytes([NotNull] this IOutgoingSocket socket, TimeSpan timeout, params byte[][] frames)
+        {
+            return TrySendMultipartBytes(socket, timeout, (IEnumerable<byte[]>)frames);
+        }
+
+        public static bool TrySendMultipartBytes([NotNull] this IOutgoingSocket socket, TimeSpan timeout,
+            IEnumerable<byte[]> frames)
+        {
+            var enumerator = frames.GetEnumerator();
```

Comment:
Note sure it matters much, but I think .NET calls `Dispose` on any `IEnumerator<T>` which is `IDisposable`. Info on [MSDN](https://msdn.microsoft.com/en-us/library/aa664754(v=vs.71).aspx).